agg-ast/schema_derivation/src/match_schema_derivation.rs

```diff
@@ -329,6 +329,7 @@ fn match_derive_schema_for_op(
         }
         MatchBinaryOp::Exists => {
             if is_exists_true_bson(b) {
+                state.result_set_schema = promote_missing(&state.result_set_schema);
```

Comment:
result set schema difference uses get_or_create_schema_for_path_mut to get field schemas to be modified, which is why this operator came up in the first place. result_set_schema_difference only removes explicitly named types, so we have to promote missing to make it visible to the function.